install-redis.sh

```diff
@@ -1,8 +1,132 @@
+        "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" |
+            tee /etc/apt/sources.list.d/docker.list >/dev/null
+    fi
+
+    apt-get -y update
+    apt-get -y install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+    dockerd >/dev/null 2>&1 </dev/null &
+    docker run hello-world
+    docker rm $(docker ps -a | grep hello | cut -d" " -f1)
+    docker rmi hello-world
+}
+
+create_redis_conf() {
+
+    if [ -f "$REDIS_CONF_FILE" ]; then
+        echo "O arquivo $REDIS_CONF_FILE já existe."
+    else
+        cat <<EOF >$REDIS_CONF_FILE
+bind 0.0.0.0 ::0
+protected-mode no
+port 6379
+tcp-backlog 511
+timeout 0
+tcp-keepalive 300
+daemonize no
+pidfile /var/run/redis_6379.pid
+loglevel notice
+logfile ""
+databases 16
+always-show-logo no
+set-proc-title yes
+proc-title-template "{title} {listen-addr} {server-mode}"
+locale-collate ""
+stop-writes-on-bgsave-error yes
+rdbcompression yes
+rdbchecksum yes
+dbfilename dump.rdb
+rdb-del-sync-files no
+dir ./
+replica-serve-stale-data yes
+replica-read-only yes
+repl-diskless-sync yes
+repl-diskless-sync-delay 5
+repl-diskless-sync-max-replicas 0
+repl-diskless-load disabled
+repl-disable-tcp-nodelay no
+replica-priority 100
+acllog-max-len 128
+lazyfree-lazy-eviction no
+lazyfree-lazy-expire no
+lazyfree-lazy-server-del no
+replica-lazy-flush no
+lazyfree-lazy-user-del no
+lazyfree-lazy-user-flush no
+oom-score-adj no
+oom-score-adj-values 0 200 800
+disable-thp yes
+appendonly no
+appendfilename "appendonly.aof"
+appenddirname "appendonlydir"
+appendfsync everysec
+no-appendfsync-on-rewrite no
+auto-aof-rewrite-percentage 100
+auto-aof-rewrite-min-size 64mb
+aof-load-truncated yes
+aof-use-rdb-preamble yes
+aof-timestamp-enabled no
+slowlog-log-slower-than 10000
+slowlog-max-len 128
+latency-monitor-threshold 0
+notify-keyspace-events ""
+hash-max-listpack-entries 512
+hash-max-listpack-value 64
+list-max-listpack-size -2
+list-compress-depth 0
+set-max-intset-entries 512
+set-max-listpack-entries 128
+set-max-listpack-value 64
+zset-max-listpack-entries 128
+zset-max-listpack-value 64
+hll-sparse-max-bytes 3000
+stream-node-max-bytes 4096
+stream-node-max-entries 100
+activerehashing yes
+client-output-buffer-limit normal 0 0 0
+client-output-buffer-limit replica 256mb 64mb 60
+client-output-buffer-limit pubsub 32mb 8mb 60
+proto-max-bulk-len 512mb
+hz 10
+dynamic-hz yes
+aof-rewrite-incremental-fsync yes
+rdb-save-incremental-fsync yes
+jemalloc-bg-thread yes
+EOF
+        echo "File $REDIS_CONF_FILE created successfully."
+    fi
+}
+
+function redis_setup() {
+    create_redis_conf
+
+    docker image pull redis/redis-stack-server:$REDIS_STACK_VERSION
+
+    docker run --name redis --restart always -v $REDIS_CONF_FILE:/redis-stack.conf --port ${REDIS_PORT}:6379 -d redis/redis-stack-server:$REDIS_STACK_VERSION
+}
+
+user_setup
+docker_setup
+redis_setup
```

Comment:
I didn't see where Redis port is being set open in ufw. Are we doing it somewhere else?
Anyway, this port should not be open for every IP, but only for the OpenFaaS server.
This setup is for standalone Redis only, right? If not I see some other missing steps here.

Reply:
Yes, this is a standalone setup. I was considering initiating a new task to provision a Redis and MongoDB cluster using Terraform.